### ISSUE TITLE
Numerify arguments for Ruby math functions

### ISSUE THREAD

**Issue by nvanoorschot** [rubysolo/dentaku#119]
Great gem!

Yet, I ran into a small inconvenience.

Since the following works fine with String arguments:
```ruby
Dentaku('3 + x', x: '5') == 8
```

I also expected this to work:
```ruby
Dentaku('MAX(3, x)', x: '5') == 5
```

But instead I get a error telling me I can't compare Integers with Strings in the max function of Ruby. Within the Arthmetic functions the arguments are converted to numbers before evaluation but not in the other functions.

I would suggest to convert all arguments to Numeric for all functions where relevant. That would be MAX, MIN, ROUND, ROUNDDOWN, ROUNDUP and the Ruby Math functions.

In my fork I got MAX and MIN working with this principle, if you like I can submit a PR when I'm done with this.

**Comment by rubysolo** [comment]
I suppose the `ROUND` and Ruby math functions only make sense for numeric inputs, but I could see using `MIN`/`MAX` on strings:

```ruby
irb> "test".split("").max
=> "t"
irb> Dentaku('MAX("A","B","C")')
=> "C"
```

However, I'm not sure that anyone is actually using it that way.  Also these functions are assumed to be numeric in spreadsheets (at least in Numbers and Google Sheets), and are registered as `:numeric` type in Dentaku, so I'll look forward to your PR.  😄

**Comment by rubysolo** [comment]
Closed by bd3a76c30f4719dcc573a141ea3977851fe6a7a0
